server.go

```diff
@@ -81,32 +79,40 @@ func RegisterProxy(instance, service, domain string, port int, host string, ips
 		}
 	}
 
-	s, err := newServer(iface)
+	entry.HostName = strings.ToLower(entry.HostName)
+
+	return entry, nil
+}
+
+// Register a service proxy by given argument. This call will skip the hostname/IP lookup and
+// will use the provided values.
+func RegisterProxy(initialEntry *ServiceEntry, entries chan *ServiceEntry, iface *net.Interface) (*Server, error) {
+	s, err := newServer(initialEntry, iface)
 	if err != nil {
 		return nil, err
 	}
 
-	s.service = entry
-	s.service.HostName = strings.ToLower(s.service.HostName)
-
-	go s.mainloop()
-	go s.probe()
+	s.waitGroup.Add(1)
+	go s.mainloop(entries)
 
 	return s, nil
 }
 
 // Server structure incapsulates both IPv4/IPv6 UDP connections
 type Server struct {
-	service        *ServiceEntry
-	ipv4conn       *net.UDPConn
-	ipv6conn       *net.UDPConn
-	shouldShutdown bool
-	shutdownLock   sync.Mutex
-	ttl            uint32
+	ipv4conn          *net.UDPConn
+	ipv6conn          *net.UDPConn
+	initialEntry      *ServiceEntry
+	shouldShutdown    bool
+	shutdownChan      chan bool
+	shutdownChanProbe chan bool
```

Comment:
It looks like this channel isn't used anywhere (could only find one reference to it commented out), can we remove this from the struct?

Reply:
good find